src/psij/web/js/lib.js

```diff
@@ -60,22 +74,44 @@ var settingsDefaults = function(dict) {
 var makeSetting = function(name, callback, type) {
     return {
         get: function() {
-            var val = Cookies.get(name);
-            if (type == "bool") {
-                return val == "true";
-            }
-            else {
-                return val;
+
+            var sbb = "none found";
+
+            if( name === 'showBranchBubbles' ) {
+                sbb = localStorage.getItem('showBranchBubbles') === 'true' ? 'true' : false;
+            } else  if( name === 'inactiveTimeout' ) {
+                sbb = localStorage.getItem('inactiveTimeout') || 10;
             }
+
+            return sbb;
```

Comment:
^^^

Reply:
Was there a need to specify default values here on top of initializing default values when the page was loaded?
What makes this code more robust, exactly?

Reply:
because that way you ensure that the value you get out is exactly one of those default values and nothing else.  the need is robustness.

Reply:
You get the default value in this function unless something is already stored in local storage.
So, if the first thing you do when you load lib.js is to store default values in local storage, will you ever get the default values from this function?

Reply:
Anyway, I don't want to drag this one, but no, the code is not more robust and this change should not have been here, since the settings are all initialize at page load. In fact, this is less robust because relies on hardcoded strings, fetches values twice for no good reason, and, for showBranchBubbles returns either the string 'true' or the boolean false, making it difficult to use in either a string or boolean expression.